### DISCUSSION TITLE
Copilot is not suggesting anything. Even the extension enabled globally.

### DISCUSSION THREAD

**Discussion by harshprakholiya** [community/community#77073] [Copilot Conversations] [answered]
## GitHub Copilot Issue in Visual Studio Code

I'm facing an issue with GitHub Copilot in Visual Studio Code, where it's not suggesting any code. Even though it says "enabled globally" in the extensions tab settings, the icon in the status bar has a diagonal line through it. When I click on the icon, it shows the status as "Ready (disabled)."

### Problem Details:

- **Symptoms:**
  - GitHub Copilot is not providing code suggestions.
  - Extension settings indicate that it's enabled globally.

- **Observations:**
  - In the extensions tab settings, it shows as "disable," That means it is enabeled.
  - Clicking on the icon in the status bar reveals the status as "Ready (disabled)."

### Confusion and Seeking Help:

I'm confused about what to do to re-enable GitHub Copilot. It seems contradictory that it's marked as enabled globally but also shows as "Ready (disabled)." How can I resolve this issue and get GitHub Copilot to work properly again?

Any guidance or suggestions on how to troubleshoot and enable GitHub Copilot would be greatly appreciated.

**Comment by Bibertv** [comment]
Got the same Problem. If you already found a solution, you might share?

**Reply by jlukss** [reply]
Not OP, but I had to press on the copilot icon and select "Enable code completion for plaintext", and it started working. Looks like it detects some of my code as plaintext and disables it.

**Comment by Akash1134** [comment]
Hi @ItsMeHarshRakholiya ,

Thanks for posting in the GitHub Community.


- First, make sure that you are using the most recent versions of VSCode and the Copilot extension. 
- Next, make sure that you have "editor.inlineSuggest.enabled": true in your Github Copilot extension settings. You can access these with the gear symbol on the extension page in your editor.

If the problem persists, please reply with (from the machine where you are having the problem):

- The output from View > Command Palette... > Github Copilot: Collect Diagnostics
- Any output you see in the bottom output panel with "Github Copilot" selected from the dropdown (while working on some code).
- Any information about the network connection/environment you are working in, like VPN/Firewall/Proxy/VM.

**Reply by harshprakholiya** [reply]
Hi @Akash1134 ,

Thank you so much for your assistance! After going through your suggestions and adding "editor.inlineSuggest.enabled":  true (which was not there in setting.json) it worked!

I appreciate your quick response and guidance. Your support in the GitHub community is invaluable!

Thanks again

**Reply by abuhassan** [reply]
I have the same problem, the copilot icon is stuck at (Disabled). I have changed the copilot json setting but there is no change. What else can I do?

**Reply by kirandash** [reply]
Go the same problem even after adding: `"editor.inlineSuggest.enabled": true` 

shared below is my diagnostics:

## Copilot

- Version: 1.171.0
- Build: prod
- Editor: vscode/1.87.0

## Environment

- http_proxy: n/a
- https_proxy: n/a
- no_proxy: n/a
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: n/a
- OPENSSL_CONF: n/a

## Feature Flags

- Custom Certificates: disabled
- Send Restricted Telemetry: enabled
- Chat: enabled

## Node setup

- Number of root certificates: 137
- Operating system: Darwin
- Operating system version: 23.1.0
- Operating system architecture: x64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: n/a
- Proxy port: n/a
- Kerberos SPN: n/a
- Reject unauthorized: disabled
- Fetcher: HelixFetcher

## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- api.githubcopilot.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy authentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: false
- Is `mac-ca` installed?: false

## Authentication

- GitHub username: kirandash

**Reply by calapurka** [reply]
Hey man, I was having the same problem and my "editor.inlineSuggest.enabled" was already set to true, but in the JSON settings of the extension, I changed the * value from false to true and it immediately worked

```
"github.copilot.enable": {
        "*": true,
        "plaintext": false,
        "markdown": false,
        "scminput": false,
        "yaml": false,
        "javascript": true
},
```

**Comment by y-nako618** [comment, marked as answer]
Just click on the copilot icon at the bottom right corner and type "Enable Completions" or "Disable Completions".

**Reply by yukunzGIT** [reply]
It works. Thanks a lot.

**Reply by obtahir** [reply]
Wait how? I don't see that option. I can only get suggestions by typing ctrl+I to open a text box for talking with copilot. But it never shows "ghost text" for me if I write any comments, code or anything

**Reply by hhhhlkf** [reply]
> Wait how? I don't see that option. I can only get suggestions by typing ctrl+I to open a text box for talking with copilot. But it never shows "ghost text" for me if I write any comments, code or anything

Just click the copliot icon in the lower right corner and type "Enable Completions" in the upper input box that pops up.

**Reply by ding9889** [reply]
Hi @obtahir, you can just click the copilot icon and you'll see a dropdown list named Github Copilot Menu in the mid-top (will replace the **_Search_**) of the editor. Then you can simply click on the option "Enable Completions for xxx" (xxx is the language you are using), and everything should work after that.

**Reply by JasonTan94** [reply]
What if I have already enabled the settings? It still just helps me complete the comment sentence that asks it to generate a certain function, but it doesn't seem to provide any code solution?

**Reply by anlagbr** [reply]
![image](https://github.com/user-attachments/assets/1db57525-8cff-4aed-87a6-23572e7a161b)

![image](https://github.com/user-attachments/assets/89e015a4-699d-475c-b038-c5f8b798738e)

**Comment by pklitscher** [comment]
For anyone who had this issue but completions was enabled, but still not getting completions. Disabling then enabling again resolved the issue for me.

**Reply by tsakalidiskostas** [reply]
Thanks, I tried all the above, no change, tried this and restarted VSCode, worked. 👍
One more day in VSC world

**Comment by Powersource** [comment]
i also had to do this in `/home/user/.vscode-oss/argv.json`

```json
	"enable-proposed-api": ["GitHub.copilot"],
```

**Reply by VanderSP** [reply]
i tried this in first place, didnt did anything on this conversation, only this you said.. and restarted and worked....

worth noting that im on VSCode Insider

**Comment by jonschlinkert** [comment]
Where is this controlled now? The icon is no longer on the bottom right, and I used it constantly.

**Reply by ding9889** [reply]
It's now next to the "Search" on the mid-top.

**Comment by cdsaenz** [comment]
For anybody who didn't find a solution here, it could also be the same thing that happened to me :)
![image](https://github.com/user-attachments/assets/9f414bd6-2198-4a62-b93c-20771a11be76)

**Reply by Cerberus322** [reply]
thanks, this was my problem xP

**Comment by anlagbr** [comment]
![image](https://github.com/user-attachments/assets/1db57525-8cff-4aed-87a6-23572e7a161b)

![image](https://github.com/user-attachments/assets/89e015a4-699d-475c-b038-c5f8b798738e)

**Comment by biansor** [comment]
None of the answers above works for me because I forgot to sign in.
